integration/helpers.go

```diff
@@ -166,6 +167,25 @@ func AssertEqualError(t testing.TB, expected mongo.CommandError, actual error) b
 	return assert.Equal(t, expected, a)
 }
 
+// AssertEqualErrorCode asserts error code, name and wrapped are the same.
+func AssertEqualErrorCode(t *testing.T, expected, actual error) bool {
```

Comment:
I think we should remove `bool` result (we could always add it later) and rename function to something like `AssertSimilarError` or `AssertMatchesError` (please check existing helpers if there is some pattern) since we don't compare just error codes anymore

Reply:
Used `AssertMatchesCommandError`. Other existing helpers just were removing Raw fields, which is different for this case.